### DISCUSSION TITLE
Could we add a kustomization.yaml file to the manifest directory?

### DISCUSSION THREAD

**Discussion by davidshen84** [k3s-io/k3s#8371] [Ideas]
I need most of the default add-ons but with minor patches. I wonder if it is safe to drop a `kustomization.yaml` file in the `/var/lib/rancher/k3s/server/manifests/` directory. Because we cannot use an *absolute path* in the `resources` property in a `kustomization.yaml` file.

Also, would k3s add support to use that `kusotomization.yaml` file to transform other yaml files in that directory?

**Comment by brandond** [comment]
no, they're not meant for direct consumption - they have templating directives in them that are rewritten by the server, for example https://github.com/k3s-io/k3s/blob/550dd0578f79882e1a78d8468fdbefa95faa145c/manifests/local-storage.yaml#L111

**Comment by davidshen84** [comment]
Got it. Thanks

**Comment by brandond** [comment]
We could probably evaluate using kustomize instead of the current templating logic, but once you get too far down that road you probably want something more configurable than we'd want to implement.
